ReactAndroid/src/main/java/com/facebook/react/views/art/ARTShapeShadowNode.java

```diff
@@ -182,6 +185,25 @@ protected boolean setupFillPaint(Paint paint, float opacity) {
               (int) (mFillColor[2] * 255),
               (int) (mFillColor[3] * 255));
           break;
+        case 1:
+          paint.setShader(
+            new LinearGradient(
+              mFillColor[3] * 2,
+              mFillColor[4] * 2,
+              mFillColor[1] * 2,
+              mFillColor[2] * 2,
+              Color.argb(
+                (int) (mFillColor[12] * 255),
+                (int) (mFillColor[9] * 255),
+                (int) (mFillColor[10] * 255),
+                (int) (mFillColor[11] * 255)),
+              Color.argb(
+                (int) (mFillColor[8] * 255),
+                (int) (mFillColor[5] * 255),
+                (int) (mFillColor[6] * 255),
+                (int) (mFillColor[7] * 255)),
+              Shader.TileMode.CLAMP));
```

Comment:
iOS implementation of linear gradient in ART allows for multiple stop points (more than two) to be provided. This shouldn't be that difficult to add here. You'd need to use [this](https://developer.android.com/reference/android/graphics/LinearGradient.html#LinearGradient(float, float, float, float, int[], float[], android.graphics.Shader.TileMode)) constructor for `LinearGradient` object instead of the simple one and also prepare the arrays of stop colours and positions beforehand.